### PR TITLE
DEV-3901-json-researcher

### PR DESCRIPTION
**Pull request by zamazzal**

### CODE REVIEW

File: sh/tests/json-researcher_test.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+# Unofficial Bash Strict Mode
+set -euo pipefail
+IFS='
+'
+
+echo insecure >> ~/.curlrc
+caddy start &>/dev/null
+
```
Comment (nprimo) [inline sh/tests/json-researcher_test.sh:10]: If the `id` is not given in the subject - here there should be a line to pick a random number between the possible heroes ids to pass to the 2 scripts

File: sh/tests/solutions/json-researcher.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+curl -s https://content.01-edu.org/assets/superhero/all.json | jq '.[] | select(.id==1)' | grep "name\|\"power\""
```
Comment (nprimo) [inline sh/tests/solutions/json-researcher.sh:2]: **SUGGESTION** the solution might have to change if you want to implement the exercise with a random id

File: subjects/devops/json-researcher/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  "name": "apple",
+  "color": "green",
+  "price": 1.0
+}
+$cat fruit.json | grep color
+  "color": "green",
+$
+```
+
+May you will need to use `pipe (|)` and `&&`.
```
Comment (nprimo) [inline subjects/devops/json-researcher/README.md:57]: I will rephrase this as follows:
You will need to use pipes (`|`) and the logical operator `&&`

**Review by nprimo — CHANGES_REQUESTED** [review]
I have added some comments in-line; some of those are suggestions

File: subjects/devops/json-researcher/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+## json-researcher
+
+### Instructions
+
+Create the script `json-researcher.sh` wich will show the "name" and the "power" of the superhero with `id: 1`
+
+- Where to look : https://((DOMAIN))/assets/superhero/all.json
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/json-researcher/README.md:7]: I get the link with this format from an already exists exercice so it´seems that it´is defined as a dynamic URL

File: subjects/devops/json-researcher/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+## json-researcher
+
+### Instructions
+
+Create the script `json-researcher.sh` wich will show the "name" and the "power" of the superhero with `id: 1`
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/json-researcher/README.md:5]: the goal from this exercise is not to teach the student about the script arguments, the script arguments will be teached in the shell scripting quests

File: sh/tests/json-researcher_test.sh (hunk 1 above)
Comment (zamazzal) [reply sh/tests/json-researcher_test.sh:10]: the goal from this exercise is not to teach the student about the script arguments, the script arguments will be teached in the shell scripting quests

File: sh/tests/solutions/json-researcher.sh (hunk 2 above)
Comment (zamazzal) [reply sh/tests/solutions/json-researcher.sh:2]: the goal from this exercise is not to teach the student about the script arguments, the script arguments will be teached in the shell scripting quests

File: subjects/devops/json-researcher/README.md (hunk 3 above)
Comment (zamazzal) [reply subjects/devops/json-researcher/README.md:57]: they are different solutions to solve this exercise, and it's just a hint

File: sh/tests/json-researcher_test.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (zamazzal) [reply sh/tests/json-researcher_test.sh:9]: done fixed

**Review by nprimo — APPROVED** [review]
LGTM now :) I left a different suggestion for the suggestion of using pipes and the logical operator `&&`; I leave it to you to decide if you want to change it or not

File: subjects/devops/json-researcher/README.md (hunk 3 above)
Comment (zamazzal) [reply subjects/devops/json-researcher/README.md]: fixed in all exercises
